From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Sun, 16 Jul 2023 14:54:08 +0200
Subject: [PATCH] Cache Block, Item and BlockState registry index - Use -
 BlockState

Replaces lookups of `BlockState` registry keys (occurrences of `Block.BLOCK_STATE_REGISTRY.getId`) with the lazily computed value in `BlockState` (`BlockState.getIndexInBlockStateRegistry`) added in the "Cache Block, Item and BlockState registry index - Provide" patch.

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

diff --git a/src/main/java/net/minecraft/world/level/block/Block.java b/src/main/java/net/minecraft/world/level/block/Block.java
index 8604af900e1230724b2288b7287367aa3fb5d316..34de48df6e4aceab651d69dbd9c03c73de212bea 100644
--- a/src/main/java/net/minecraft/world/level/block/Block.java
+++ b/src/main/java/net/minecraft/world/level/block/Block.java
@@ -197,7 +197,7 @@ public class Block extends BlockBehaviour implements ItemLike {
         if (state == null) {
             return 0;
         } else {
-            int i = Block.BLOCK_STATE_REGISTRY.getId(state);
+            int i = state.getIndexInBlockStateRegistry(); // Fiddle - cache registry indices as fields - use instead of lookup - block state
 
             return i == -1 ? 0 : i;
         }
